File: 7/main.java
class Solution {
    public boolean isSubset(int a[], int b[]) {
        int n = a.length;
        int m = b.length;
        // Create a hash map to store the frequencies of elements in array a
        HashMap<Integer, Integer> hm = new HashMap<>();

        // Increment the frequency of each element in array a
        for (int i = 0; i < n; i++) {
            hm.put(a[i], hm.getOrDefault(a[i], 0) + 1);
        }

        // Check if each element in array b is present in array a and decrement its
        // frequency
        for (int i = 0; i < m; i++) {
            if (hm.containsKey(b[i])) {
                // If frequency becomes 1, remove the element from the hash map
                if (hm.get(b[i]) == 1) {
                    hm.remove(b[i]);
                } else {
                    // Decrement the frequency of the element
                    hm.put(b[i], hm.get(b[i]) - 1);
                }
            } else {
                // If an element in array b is not present in array a, return "No"
                return false;
            }
        }

        // If all elements in array b are present in array a, return "Yes"
        return true;
    }
}

File: 7/main.py
from collections import defaultdict


class Solution:
    #Function to check if a is a subset of b.
    def isSubset(self, a, b):
        #creating two dictionaries to store the frequency of each element in a and b.
        dic1 = defaultdict(lambda: 0)

        #iterating over a and updating the frequency in dic1.
        for i in a:
            dic1[i] += 1

        #iterating over b and checking if the frequency of each element in b is greater than 0 in dic1.
        for i in b:
            if dic1[i] > 0:
                dic1[i] -= 1
            else:
                return False

        return True

